### PR TITLE
QP-3 - Add ability for admin to create/upload new questions (server-side).

### PR DESCRIPTION
**Pull request by jrrakh**
**QP-3** - Add ability for admin to create/upload new questions (server-side).

- Add transaction manager.
- Add ability to upload xml with questions.
- Update some entities for JAXB.
- Update 'QuizFacade' - now link all answers and comments for questions.
- Remove 'AuthenticationService', because we use 'AuthenticationProvider'.

### CODE REVIEW

File: src/main/java/com/testportal/impl/facade/QuizFacade.java [hunk 1]
```diff
@@ -49,4 +46,62 @@ public boolean createQuestionsFromXML(File input) {
...
+        questions.forEach(question -> {
+            if (CollectionUtils.isNotEmpty(question.getAnswerList())) {
+                answers.addAll(question.getAnswerList());
+            }
+        });
+        return answers;
+    }
+
+    private void setIdToAnswersAndComments(List<Question> questions) {
+        questions.stream()
```
Comment (Ghost139) [inline src/main/java/com/testportal/impl/facade/QuizFacade.java]: As i know we can use forEach without stream

File: src/main/java/com/testportal/impl/facade/QuizFacade.java [hunk 2]
```diff
@@ -49,4 +46,62 @@ public boolean createQuestionsFromXML(File input) {
...
+                    if (CollectionUtils.isNotEmpty(answers)) {
+                        answers.forEach(answer -> {
+                            if (answer != null) {
+                                answer.setQuestionId(question.getId());
+                            }
+                        });
+                    }
+                });
+
+        questions.stream()
```
Comment (Ghost139) [inline src/main/java/com/testportal/impl/facade/QuizFacade.java]: Same as above

File: src/main/java/com/testportal/impl/facade/QuizFacade.java (hunk 1 above)
Comment (jrrakh) [inline src/main/java/com/testportal/impl/facade/QuizFacade.java]: Yes, I'll change this.

File: src/main/java/com/testportal/impl/facade/QuizFacade.java (hunk 2 above)
Comment (jrrakh) [inline src/main/java/com/testportal/impl/facade/QuizFacade.java]: Yes, I'll change this.

File: src/main/java/com/testportal/impl/facade/QuizFacade.java (hunk 1 above)
Comment (jrrakh) [inline src/main/java/com/testportal/impl/facade/QuizFacade.java]: Done.

File: src/main/java/com/testportal/impl/facade/QuizFacade.java (hunk 2 above)
Comment (jrrakh) [inline src/main/java/com/testportal/impl/facade/QuizFacade.java]: Done.

**Pull request merged**